### PR TITLE
add bindp and noshortcutsinhibit

### PR DESCRIPTION
**Pull request by MightyPlaza**
adds `bindp` and `noshorcutsinhibit` window rule to allow control over https://wayland.app/protocols/keyboard-shortcuts-inhibit-unstable-v1 per bind and per client
using `bindp` because all the good letters have already been used

seems to work fine on qemu
wiki PR: https://github.com/hyprwm/hyprland-wiki/pull/724

### CODE REVIEW

File: src/protocols/ShortcutsInhibit.cpp [hunk 1]
```diff
@@ -73,6 +73,9 @@ bool CKeyboardShortcutsInhibitProtocol::isInhibited() {
     if (!g_pCompositor->m_pLastFocus)
         return false;
 
+    if (g_pCompositor->getWindowFromSurface(g_pCompositor->m_pLastFocus.lock())->m_sWindowData.noShortcutsInhibit.valueOrDefault())
```
Comment (vaxerski) [inline src/protocols/ShortcutsInhibit.cpp:76]: ehhhh is this a good idea?

Comment (MightyPlaza) [reply src/protocols/ShortcutsInhibit.cpp:76]: yeah you're probably right
what's the solution? storing a `PHLWINDOWREF` on each `CWLSurface` when created or will that cause issues?

Comment (vaxerski) [reply src/protocols/ShortcutsInhibit.cpp:76]: not my point, the question is should the state of isInhibited() be altered by the focus. IMO no

Comment (MightyPlaza) [reply src/protocols/ShortcutsInhibit.cpp:76]: it already is doe
the next bit of the function is
```cpp
for (auto& in : m_vInhibitors) {
     if (in->surface() != g_pCompositor->m_pLastFocus)
         continue;
```
which already depends on focus

Comment (vaxerski) [reply src/protocols/ShortcutsInhibit.cpp:76]: ah, right. It's fine then

**Review by vaxerski — APPROVED** [review]
cool

**Comment by MightyPlaza** [comment]
just to leave here, I initially thought the concern was the unnecessary looping over all windows for each bind on each key press

**Comment by vaxerski** [comment]
it's not there, read the source for the func
I fixed it an hour ago

**Pull request merged**